src/viam/module/module.py

```diff
@@ -83,17 +82,21 @@ async def _connect_to_parent(self):
                 ),
             )
 
-    async def _get_component(self, name: ResourceName) -> ComponentBase:
+    async def _get_resource(self, name: ResourceName) -> ResourceBase:
         await self._connect_to_parent()
         assert self.parent is not None
         await self.parent.refresh()
-        return self.parent.get_component(name)
-
-    async def _get_dependencies(self, dependencies: Sequence[str]) -> Mapping[ResourceName, ComponentBase]:
-        deps: Mapping[ResourceName, ComponentBase] = {}
+        if name.type == RESOURCE_TYPE_COMPONENT:
+            return self.parent.get_component(name)
+        elif name.type == RESOURCE_TYPE_SERVICE:
+            return self.parent.get_service(name)
+        raise ValueError("Dependency does not describe a component nor a service")
+
+    async def _get_dependencies(self, dependencies: Sequence[str]) -> Mapping[ResourceName, ResourceBase]:
+        deps: Mapping[ResourceName, ResourceBase] = {}
         for dep in dependencies:
             rn = resource_name_from_string(dep)
-            component = await self._get_component(rn)
+            component = await self._get_resource(rn)
             deps[rn] = component
```

Comment:
was originally gonna say that the `component` variable name is wrong now, but maybe this can just be one line:

```suggestion
            deps[rn] = await self._get_resource(rn)
```